logic/transaction_pool.js

```diff
@@ -495,41 +495,48 @@ TransactionPool.prototype.reindexQueues = function() {
 TransactionPool.prototype.processBundled = function(cb) {
 	const bundled = self.getBundledTransactionList(true, self.bundleLimit);
 
-	async.eachSeries(
-		bundled,
-		(transaction, eachSeriesCb) => {
-			if (!transaction) {
-				return setImmediate(eachSeriesCb);
-			}
-			library.balancesSequence.add(balancesSequenceCb => {
-				__private.processVerifyTransaction(transaction, true, err => {
-					// Remove bundled transaction after asynchronous processVerifyTransaction to avoid race conditions
-					self.removeBundledTransaction(transaction.id);
-					// Delete bundled flag from transaction so it is qualified as "queued" in queueTransaction
-					delete transaction.bundled;
-
-					if (err) {
-						library.logger.debug(
-							`Failed to process / verify bundled transaction: ${
-								transaction.id
-							}`,
-							err
-						);
-						return setImmediate(balancesSequenceCb);
+	library.balancesSequence.add(
+		balancesSequenceCb => {
+			async.eachSeries(
+				bundled,
+				(transaction, eachSeriesCb) => {
+					if (!transaction) {
+						return setImmediate(eachSeriesCb);
```

Comment:
If we filter the array before star each loop we can avoid this condition. 

`bundled = bundled.filter(Boolean);`

Reply:
I am not refactoring code right now. So, didn't make this change here.